Detectors/EMCAL/reconstruction/src/Clusterizer.cxx

```diff
@@ -0,0 +1,180 @@
+            getClusterFromNeighbours(clusterDigits, row + rowDiff, column + colDiff);
+            // Add the digit to the current cluster -- if we end up here, the selected cluster fulfills the condition
+            clusterDigits.emplace_back(mDigitMap[row + rowDiff][column + colDiff]);
+          }
+  }
+}
+
+///
+/// Get row (phi) and column (eta) of a digit, values corresponding to topology
+///
+//____________________________________________________________________________
+void Clusterizer::getTopologicalRowColumn(const Digit& digit, int& row, int& column)
+{
+  // Get SM number and relative row/column for SM
+  auto cellIndex = mEMCALGeometry->GetCellIndex(digit.getTower());
+  int nSupMod = std::get<0>(cellIndex);
+
+  auto phiEtaIndex = mEMCALGeometry->GetCellPhiEtaIndexInSModule(nSupMod, std::get<1>(cellIndex), std::get<2>(cellIndex), std::get<3>(cellIndex));
+  row = std::get<0>(phiEtaIndex);
+  column = std::get<1>(phiEtaIndex);
+
+  // Add shifts wrt. supermodule and type of calorimeter
+  // NOTE:
+  // * Rows (phi) are arranged that one space is left empty between supermodules in phi
+  //   This is due to the physical gap that forbids clustering
+  // * For DCAL, there is an additional empty column between two supermodules in eta
+  //   Again, this is to account for the gap in DCAL
+
+  row += nSupMod / 2 * (24 + 1);
+  // In DCAL, leave a gap between two SMs with same phi
+  if (!mEMCALGeometry->IsDCALSM(nSupMod)) // EMCAL
+    column += nSupMod % 2 * 48;
+  else
+    column += nSupMod % 2 * (48 + 1);
+}
+
+///
+/// Return number of found clusters. Start clustering from highest energy cell.
+//____________________________________________________________________________
+void Clusterizer::findClusters(const std::vector<Digit>& digitArray)
+{
+  mFoundClusters.clear();
+  mDigitIndices.clear();
+
+  // Algorithm
+  // - Fill digits in 2D topological map
+  // - Fill struct arrays (energy,x,y)  (to get mapping energy -> (x,y))
+  // - Create 2D bitmap (digit is already clustered or not)
+  // - Sort struct arrays with descending energy
+  //
+  // - Loop over arrays:
+  // --> Check 2D bitmap (don't use digit which are already clustered)
+  // --> Take valid digit with highest energy as seed (they are already sorted)
+  // --> Recursive to neighboughs and create cluster
+  // --> Seed cell and all neighbours belonging to cluster will be put in 2D bitmap
+
+  // Reset digit maps and cell masks
+  // Loop over one array dim, then reset each array
+  for (auto iArr = 0; iArr < NROWS; iArr++) {
+    mCellMask[iArr].fill(0);
+    mDigitMap[iArr].fill(0);
+  }
+
+  // Calibrate digits and fill the maps/arrays
+  int nCells = 0;
+  double ehs = 0.0;
+  for (auto dig : digitArray) {
+    Float_t digitEnergy = dig.getEnergy();
+    Float_t time = dig.getTimeStamp();
+
+    if (digitEnergy < mThresholdCellEnergy || time > mTimeMax || time < mTimeMin)
```

Comment:
please use {} even for 1-liners

Reply:
Done, I tried to find all occurrences in the commit.